fog/view/server/src/fog_view_router_service.rs

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+use futures::{future::try_join_all, FutureExt, SinkExt, TryFutureExt, TryStreamExt};
+use grpcio::{DuplexSink, RequestStream, RpcContext, WriteFlags};
+use mc_attest_api::attest;
+use mc_common::logger::{log, Logger};
+use mc_fog_api::{
+    view::{FogViewRouterRequest, FogViewRouterResponse},
+    view_grpc::FogViewRouterApi,
+};
+use mc_fog_uri::FogViewStoreUri;
+use mc_fog_view_enclave_api::ViewEnclaveProxy;
+use mc_util_grpc::{rpc_logger, rpc_permissions_error};
+use mc_util_metrics::SVC_COUNTERS;
+use std::sync::Arc;
+
+#[derive(Clone)]
+pub struct FogViewRouterService<E: ViewEnclaveProxy> {
+    enclave: E,
+    shards: Vec<Arc<FogViewStoreUri>>,
```

Comment:
That's a good point. In future PRs, we'll need to track both connections to load balancers and the individual stores. As such, implementing it will be a larger change, so I'll just take care of that in one of the later PRs. Thanks for the suggestion.